kuma/wiki/feeds.py

```diff
@@ -154,21 +156,28 @@ def get_object(self, request, format, tag=None, category=None):
                 reverse('kuma.wiki.views.list_documents'))
 
     def items(self):
-        locale = ((self.request.GET.get('all_locales', False) is False)
-                  and self.request.locale or None)
-        return (Document.objects
-                        .filter_for_list(tag_name=self.tag,
-                                         category=self.category,
-                                         locale=locale)
-                        .filter(current_revision__isnull=False)
-                        .prefetch_related('current_revision',
-                                          'current_revision__creator')
-                        .order_by('-current_revision__created')
-                [:MAX_FEED_ITEMS])
+        # Temporarily storing the selected documents PKs in a list
+        # to speed up retrieval (max MAX_FEED_ITEMS size)
+        item_pks = (Document.objects
+                            .filter_for_list(tag_name=self.tag,
+                                             category=self.category,
+                                             locale=self.locale)
+                            .filter(current_revision__isnull=False)
+                            .order_by('-current_revision__created')
+                            .values_list('pk', flat=True)[:MAX_FEED_ITEMS])
+        return (Document.objects.filter(pk__in=list(item_pks))
+                                .defer('html')
+                                .prefetch_related('current_revision',
+                                                  'current_revision__creator',
+                                                  'tags'))
```

Comment:
I don't understand what difference this makes? Why is the code being removed bad?